lib/solar.js

```diff
@@ -147,6 +150,26 @@ function sin_elevation(ms, lat, lon) {
   return sin_lat * sin_dec + cos_lat * cos_dec * Math.cos(hour_angle);
 }
 
+// NREL spa.c#topocentric_azimuth_angle_astro
+// https://midcdmz.nrel.gov/sampa/
+// https://midcdmz.nrel.gov/spa/spa.h
+function azimuth(ms, lat, lon) {
+  const t = julian.to(ms);
+  const sin_lat = Math.sin(lat * RAD);
+  const cos_lat = Math.cos(lat * RAD);
+  const hour_angle = sidereal.local(t, lon * RAD) - _right_ascension(t);
+  const sin_hour_angle = Math.sin(hour_angle);
+  const cos_hour_angle = Math.cos(hour_angle);
+  const dec = _declination(t);
+  const tan_dec = Math.tan(dec);
```

Comment:
This is the only real thing I dislike. It's fine for a first-pass, but once you got it working, did you test the alternative way?

```javascript
const sin_dec = _sin_declination(t);
const tan_dec = sin_dec / Math.sqrt(1 - sin_dec * sin_dec);
```

Reply:
Also you should print this Wikipedia page out and hang it by your desk: https://en.wikipedia.org/wiki/List_of_trigonometric_identities

Reply:
That passed the tests, too!